lib/include/aws_tls.h

```diff
@@ -31,6 +31,20 @@
     #error "include FreeRTOS.h must appear in source files before include aws_tls.h"
 #endif
 
+/**
+ * @defgroup TlsErrors TLS Error Codes
+ * @brief Error codes returned by the TLS API.
+ *
+ * Note that TLS API may also propagate port-specific
+ * error codes, or codes from mbedTLS.
+ */
+/**@{ */
+#define TLS_ERROR_HANDSHAKE_FAILED    ( -2001 )   /*!< Error in handshake. */
+#define TLS_ERROR_RNG                 ( -2002 )   /*!< Error in RNG. */
```

Comment:
Pending the above, is the implication that the -20XX error code namespace is reserved for this library?

Reply:
As discussed, there is no guarantee that we are not conflicting with a device library.   Sockets errors are smaller (0-200, -10xx), MQTT, defender use an enum (so they will be smaller), OTA uses a combination of high byte lib error + lower byte driver errors,  Crypto & GG use true/false.